pkg/scheduler/schedule_one.go

```diff
@@ -159,7 +160,7 @@ func (sched *Scheduler) schedulingCycle(
 			nominatingInfo = clearNominatedNode
 			reason = v1.PodReasonSchedulerError
 		}
-		return ScheduleResult{nominatingInfo: nominatingInfo, reason: reason}, podInfo, err
+		return ScheduleResult{nominatingInfo: nominatingInfo, reason: reason}, podInfo, framework.AsStatus(err)
```

Comment:
When reason is `PodReasonSchedulerError`, return framework.AsStatus(), when reason is `PodReasonUnschedulable`, return framework.NewStatus(Unschedulable). Then we can tell the reason correctly from status.

Reply:
In `schedulingCycle`, `SchedulePod` will return some fitError, like `0/1 nodes are available: 1 node(s) didn't have free ports for the requested pod ports.`
In this scenario, reason will still be `PodReasonUnschedulable`,  and the pod will be look like
```
Reason:  "Unschedulable",
Message:  "0/1 nodes are available: 1 node(s) didn't have free ports for the requested pod ports.",
```
If we return a `framework.NewStatus(Unschedulable)` here, we'll lose the origin fitError which should be handled in `handleSchedulingFailure`
How can we handle these scenarios?

Reply:
I think we can add a WithError() method to Status. Then it will look like `framework.NewStatus(Unschedulable).WithError(fitError)`.

Reply:
This looks the same as `func AsStatus(err error) *Status` ,won't this be a bit repetitive？

Reply:
I think they're still different. AsStatus used to wrap an Error status, but we need a unschedulable one with fitError.